mithril-stm/src/stm.rs

```diff
@@ -278,22 +295,12 @@ impl StmInitializer {
         self.pk
     }
 
-    /// Build the `avk` for the given list of parties.
-    ///
-    /// Note that if this StmInitializer was modified *between* the last call to `register`,
-    /// then the resulting `StmSigner` may not be able to produce valid signatures.
-    ///
-    /// Returns an `StmSigner` specialized to
-    /// * this `StmSigner`'s ID and current stake
-    /// * this `StmSigner`'s parameter valuation
-    /// * the `avk` as built from the current registered parties (according to the registration service)
-    /// * the current total stake (according to the registration service)
-    /// # Error
-    /// This function fails if the initializer is not registered.
-    pub fn new_signer<D: Digest + Clone>(
-        self,
+    /// Function that checks whether the initializer is registered.
+    /// Returns the merkle tree index of the party if registered.
+    pub fn check_initializer<D: Digest + Clone>(
+        &self,
         closed_reg: ClosedKeyReg<D>,
```

Comment:
You are consuming the `closed_reg` for checking whether the initializer is registered. It does not look right.